vmdb/app/models/mixins/container_provider_mixin.rb

```diff
@@ -0,0 +1,42 @@
+module ContainerProviderMixin
+  extend ActiveSupport::Concern
+
+  included do
+
+    has_many :container_nodes,                      :foreign_key => :ems_id, :dependent => :destroy
+    has_many :container_groups,                     :foreign_key => :ems_id, :dependent => :destroy
+    has_many :container_services,                   :foreign_key => :ems_id, :dependent => :destroy
+
+    def self.raw_api_endpoint(hostname, port)
```

Comment:
So, class method should not be in the `included` section.  Create a `module ClassMethods` and define this as an instance method in there.  See [here](http://api.rubyonrails.org/classes/ActiveSupport/Concern.html).